### PR TITLE
SciPy LowLevelCallable from Lambdify

### PR DESCRIPTION
**Pull request by isuruf**
cc @BlackTeaAndCoffee

This creates a `LowLevelCallable` that can be used by scipy integrate functions without the overhead of python.

### CODE REVIEW

File: symengine/lib/symengine_wrapper.pyx [hunk 1]
```diff
@@ -4579,6 +4604,16 @@ cdef class LambdaDouble(_Lambdify):
     cpdef unsafe_complex(self, double complex[::1] inp, double complex[::1] out, int inp_offset=0, int out_offset=0):
         self.lambda_double_complex[0].call(&out[out_offset], &inp[inp_offset])
 
+    cpdef as_scipy_low_level_callable(self):
+        from scipy import LowLevelCallable
```
Comment (bjodah) [inline symengine/lib/symengine_wrapper.pyx]: is the import of LowLevelCallable needed here?

File: symengine/lib/symengine_wrapper.pyx [hunk 2]
```diff
@@ -4592,8 +4627,18 @@ IF HAVE_SYMENGINE_LLVM:
         cpdef unsafe_real(self, double[::1] inp, double[::1] out, int inp_offset=0, int out_offset=0):
             self.lambda_double[0].call(&out[out_offset], &inp[inp_offset])
 
+        cpdef as_scipy_low_level_callable(self):
+            from scipy import LowLevelCallable
```
Comment (bjodah) [inline symengine/lib/symengine_wrapper.pyx]: same here

**Comment by bjodah** [comment]
This is really exciting and impressive (as always). I haven't used scipy's LowLevelCallable myself yet (I have been meaning to look into it). Right now I'm quite busy, but I can take a closer look at this PR this weekend.

File: symengine/lib/symengine_wrapper.pyx [hunk 3]
```diff
@@ -4560,6 +4560,31 @@ cdef class _Lambdify(object):
...
+    cdef double _scipy_callback_llvm_real(int n, double *x, void *user_data) nogil:
+        cdef symengine.LLVMDoubleVisitor* lamb = <symengine.LLVMDoubleVisitor *>user_data
+        cdef double result
+        deref(lamb).call(&result, x)
+        return result
+
+
+def create_low_level_callable(lambdify, *args):
+    from scipy import LowLevelCallable
+    class LambdifyLowLevelCallable(LowLevelCallable):
```
Comment (bjodah) [inline symengine/lib/symengine_wrapper.pyx]: Does this class need to be closure? Or could it be defined in a try/except block around `from scipy import LowLevelCallable`?

Comment (isuruf) [reply symengine/lib/symengine_wrapper.pyx]: I didn't want to add the initialization time of scipy to symengine. That's why this is done inside a function.

**Comment by bjodah** [comment]
Looks good to me. Too bad the LowLevelCallable isn't used much in scipy at the moment (that might change though).

**Comment by isuruf** [comment]
Thanks for the review

**Comment by certik** [comment]
Very impressive. Thanks @isuruf !

**Comment by BlackTeaAndCoffee** [comment]
Hi i get the following error, when i run your simple code above. 

```
Traceback (most recent call last):
  File "foo.py", line 6, in <module>
    lmb = se.lambdify(args, [se.exp(-x*t)/t**5], as_scipy=True)
TypeError: lambdify() got an unexpected keyword argument 'as_scipy'
```

i have
```
>>> import symengine
>>> symengine__version__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'symengine__version__' is not defined
>>> symengine.__version__
'0.3.0'
>>> 
```

Best wishes Mustafa

**Comment by isuruf** [comment]
@BlackTeaAndCoffee, you'll have to use the latest version of symengine.py and symengine for this

**Pull request merged**